backends/ebpf/psa/externs/ebpfPsaHashAlgorithm.cpp

```diff
@@ -248,5 +253,141 @@ void CRC32ChecksumAlgorithm::emitGlobals(CodeBuilder* builder) {
+                builder->append("(");
+                visitor->visit(field);
+                builder->appendFormat(" >> %d)", bitsToRead);
+            }
+
+            if (remainingBits == 0) {
+                remainingBits = 16;
+                builder->endOfStatement(true);
+
+                // update checksum
+                builder->target->emitTraceMessage(builder, "InternetChecksum: word=0x%llx",
+                                                  1, tmpVar.c_str());
+                builder->emitIndent();
+                if (addData) {
+                    builder->appendFormat("%s = csum16_add(%s, %s)", stateVar.c_str(),
+                                          stateVar.c_str(), tmpVar.c_str());
+                } else {
+                    builder->appendFormat("%s = csum16_sub(%s, %s)", stateVar.c_str(),
+                                          stateVar.c_str(), tmpVar.c_str());
+                }
+                builder->endOfStatement(true);
+            }
+        }
+    }
+
+    builder->target->emitTraceMessage(builder, "InternetChecksum: new state=0x%llx",
+                                      1, stateVar.c_str());
+    builder->blockEnd(true);
+}
+
+void InternetChecksumAlgorithm::emitGlobals(CodeBuilder* builder) {
+    builder->appendLine("inline u16 csum16_add(u16 csum, u16 addend) {\n"
+                        "    u16 res = csum;\n"
+                        "    res += addend;\n"
+                        "    return (res + (res < addend));\n"
+                        "}\n"
+                        "inline u16 csum16_sub(u16 csum, u16 addend) {\n"
+                        "    return csum16_add(csum, ~addend);\n"
+                        "}");
+}
+
+void InternetChecksumAlgorithm::emitVariables(CodeBuilder* builder,
+                                              const IR::Declaration_Instance* decl) {
+    (void) decl;
+    stateVar = program->refMap->newName(baseName + "_state");
+    builder->emitIndent();
+    builder->appendFormat("u16 %s = 0", stateVar.c_str());
+    builder->endOfStatement(true);
+}
+
+void InternetChecksumAlgorithm::emitClear(CodeBuilder* builder) {
+    builder->emitIndent();
+    builder->appendFormat("%s = 0", stateVar.c_str());
+    builder->endOfStatement(true);
+}
+
+void InternetChecksumAlgorithm::emitAddData(CodeBuilder* builder,
+                                            const argumentsList & arguments) {
+    updateChecksum(builder, arguments, true);
+}
+
+void InternetChecksumAlgorithm::emitGet(CodeBuilder* builder) {
+    builder->appendFormat("((u16) (~%s))", stateVar.c_str());
+}
+
+void InternetChecksumAlgorithm::emitSubtractData(CodeBuilder* builder,
+                                                 const argumentsList & arguments) {
+    updateChecksum(builder, arguments, false);
+}
+
+void InternetChecksumAlgorithm::emitGetInternalState(CodeBuilder* builder) {
+    builder->append(stateVar);
+}
+
+// FIXME: works for constant value, but might not for other cases
```

Comment:
maybe you can move this comment closer to the line which has to be fixed?

Reply:
That was old comment.